Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderBuilderUtility.cpp

```diff
@@ -814,6 +815,51 @@ namespace AZ
                 return success;
             }
 
+            IncludedFilesParser::IncludedFilesParser()
+            {
+                AZStd::regex regex(R"(#\s*include\s+[<|"]([\w|/|\\|\.|-]+)[>|"])", AZStd::regex::ECMAScript);
```

Comment:
I'm not sure the full list of characters that could be supported, "-" was just an example of something that was missing. You might consider making the regex overly permissive to avoid accidentally filtering out some include files. You make make it allow any character that is not "<>\"\n" in the file path.

Reply:
I think the list is comprehensive enough. Beyond that It's only symbols, etc. I considered that originally but it's hard to control the greediness like <(.+)> and for whatever reason std::regex is not as powerful as python's regex.